src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -330,6 +330,7 @@ protected virtual bool IsAttributeConditionallyOmitted(NamedTypeSymbol attribute
                         }
 
                         BoundExpression boundNamedArgument = BindNamedAttributeArgument(argument, attributeType, diagnostics);
+                        this.Compilation.VariableUsePass.Visit(boundNamedArgument);
```

Comment:
>  so eliminating the warning isn't needed.

If that is the design choice you are making, that would be fine. However, this doesn't answer the question that I asked.